### ISSUE TITLE
GCop659 false alarm

### ISSUE THREAD

**Issue by sungam3r** [Geeksltd/GCop#289]
```csharp
   public static void Test()
        {
            var regex = new Regex("blablabla");
            foreach (Match match in regex.Matches("bbb")) // GCop659 Use 'var' instead of explicit type
            {
                var group = match.Groups["ccc"];
                if (group == null)
                    Console.WriteLine("xxx");
            }
        }
```

After quick fix:

```csharp
public static void Test()
        {
            var regex = new Regex("blablabla");
            foreach (var match in regex.Matches("bbb"))
            {
                var group = match.Groups["ccc"]; // CS1061	'object' does not contain a definition for 'Groups' and no accessible extension method 'Groups' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)	
                if (group == null)
                    Console.WriteLine("xxx");
            }
        }
```

GCop659 must take into account duck typing.

**Comment by watsonsong** [comment]
I met the same problem when using the XMLDocument~
